test/common/object_pool_test.cpp

```diff
@@ -54,24 +89,37 @@ TEST(ObjectPoolTests, ConcurrentCorrectnessTest) {
   TestThreadPool thread_pool;
   // This should have no bearing on the correctness of test
   const uint64_t reuse_limit = 100;
-  common::ObjectPool<ObjectPoolTestType> tested(reuse_limit);
+  common::ObjectPool<ObjectPoolTestType> tested(reuse_limit, reuse_limit);
   auto workload = [&](uint32_t tid) {
+    std::uniform_int_distribution<uint64_t> size_dist_(1, reuse_limit);
+
     // Randomly generate a sequence of use-free
     std::default_random_engine generator;
     // Store the pointers we use.
     std::vector<ObjectPoolTestType *> ptrs;
-    auto allocate = [&] { ptrs.push_back(tested.Get()->Use(tid)); };
+    auto allocate = [&] {
+      try {
+        ObjectPoolTestType *temp = tested.Get();
+        ptrs.push_back(temp->Use(tid));
+      } catch (common::NoMoreObjectException) {
+      }
```

Comment:
Do you really want to swallow the exception here? If the answer is yes, add a comment explaining why.